src/class.cc

```diff
@@ -129,7 +129,7 @@ Thumbnailer::Status Thumbnailer::GenerateAnimation(WebPData* const webp_data) {
     int middle = (min_quality + max_quality) / 2;
 
     config_.quality = middle;
-    WebPDataClear(webp_data);
+    WebPDataClear(&new_webp_data);
```

Comment:
I think there is an issue. `WebPDataClear()` must be called on each `WebPData` generated by `GenerateAnimationNoBudget()` except the one that will be output as `webp_data`.

With this fix, all `WebPData` except the last one will be cleared, which would work only if `new_webp_data.size <= byte_budget_` is true during the last `while (min_quality <= max_quality)` loop.

As a reminder:

- `WebPDataInit()` resets the `WebPData` object to an empty state, without freeing any memory,
- `WebPDataClear()` frees the memory referenced by the `WebPData` object,
- Copying a `WebPData` (with operator `=`) adds just a reference to the pointed `bytes` and does not deallocate nor allocate any memory; the previous content of the copy is leaked if not cleared beforehand and only the original or the copy must be cleared afterwards. If either the original or the copy is cleared, the other one becomes invalid (points to no longer existing memory). To deeply duplicate the data and allocate another `WebPData` object, use `WebPDataCopy()` instead (not needed here though).